axelor-account/src/main/java/com/axelor/apps/account/service/moveline/MoveLineToolServiceImpl.java

```diff
@@ -457,6 +457,14 @@ public boolean isMoveLineTaxAccount(MoveLine moveLine) {
             .orElse(""));
   }
 
+  @Override
+  public boolean isMoveLineTaxAccountOrNonDeductibleTax(MoveLine moveLine) {
+    return this.isMoveLineTaxAccount(moveLine)
+        || (!this.isMoveLineTaxAccount(moveLine)
+            && taxService.isNonDeductibleTaxesSet(moveLine.getTaxLineSet())
+            && moveLine.getIsNonDeductibleTax());
```

Comment:
looks like the logic could be simplified, `this.isMoveLineTaxAccount(moveLine)` is appearing twice.

Reply:
Yes of course :
A+(not(A) ⋅B⋅C) = A+(B⋅C)